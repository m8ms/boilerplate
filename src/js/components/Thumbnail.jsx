import React from 'react';
import { Link } from 'react-router-dom';

class Gallery extends React.Component {

    constructor(props) {
        super(props);
    }

    getThumbnailSrc() {
        const size = 'b';
        return 'https://i.imgur.com/' + (this.props.image.cover || this.props.image.id) + size + '.jpg'
    }

    render() {
        let content;

        if (this.props.noMore) {
            content = <span className="thumbnail__no-more">- Fin -</span>

        }else if (this.props.children){
            content = this.props.children;
        } else {
            content = (
                <Link to={`/image/${this.props.image.id}`}>
                    <img className="thumbnail__image" src={this.getThumbnailSrc()} alt={this.props.image.id}/>
                </Link>
            )
        }


        return (
            <div className="thumbnail">
                <div className="thumbnail__content">
                    {content}
                </div>
            </div>
        )
    }
}

export default Gallery;
