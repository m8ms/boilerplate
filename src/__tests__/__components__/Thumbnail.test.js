import React from 'react';
import Thumbnail from '../../js/components/Thumbnail';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly with children', () => {
    const tree = renderer
        .create(<Thumbnail><span>child</span></Thumbnail>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with item', () => {
    const item = {
        id: 'GRDXXdF',
        title: null,
        description: 'Sorry I haven\'t been uploading stuff like this lately. Hopefully this\'ll make up for it.',
        datetime: 1523224179,
        type: 'image/png',
        animated: false,
        width: 6000,
        height: 7000,
        size: 263782,
        views: 3965,
        tags: [],
        link: 'https://i.imgur.com/GRDXXdF.png',
        ups: 34,
        downs: 2,
        points: 32,
        score: 2
    };

    const tree = renderer
        .create(<Router><Thumbnail item={item}/></Router>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
