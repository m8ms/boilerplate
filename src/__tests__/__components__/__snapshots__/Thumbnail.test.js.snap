// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with children 1`] = `
<div
  className="thumbnail"
>
  <div
    className="thumbnail__content"
  >
    <span>
      child
    </span>
  </div>
</div>
`;

exports[`renders correctly with item 1`] = `
<div
  className="thumbnail"
>
  <div
    className="thumbnail__content"
  >
    <a
      href="/image/GRDXXdF"
      onClick={[Function]}
    >
      <img
        alt="GRDXXdF"
        className="thumbnail__image"
        src="https://i.imgur.com/GRDXXdFb.jpg"
      />
      <div
        className="thumbnail__info"
      >
        image
      </div>
    </a>
  </div>
</div>
`;
